src/Database/LSMTree/Normal.hs

```diff
@@ -251,7 +254,15 @@ updates ::
   => V.Vector (k, Update v blob)
   -> TableHandle m k v blob
   -> m ()
-updates = undefined
+updates es (TableHandle th) = do
+    -- make sure not to insert any blobs into monoidal tables
+    when (isJust (Internal.confResolveMupsert (Internal.tableConfig th))) $
+      throwIO Internal.ErrExpectedNormalTable
+    Internal.updates Internal.constResolveMupsert (V.mapStrict serialiseEntry es) th
```

Comment:
I think passing `Internal.constResolveMupsert` here is not necessary: we already checked that `Internal.confResolveMupsert` is `Nothing,` and `Internal.updates` can read the `confResolveMupsert` from the table handle using `resolveMupsert`

Reply:
Ah, yes. It's redundant.